packages/js/src/ai-assessment-fixes/components/ai-assessment-fixes-button.js

```diff
@@ -31,6 +32,14 @@ const AIAssessmentFixesButton = ( { id, isPremium } ) => {
 	const focusElementRef = useRef( null );
 	const [ buttonClass, setButtonClass ] = useState( "" );
 
+	// Only enable the button when the editor is in visual mode and all blocks are in visual mode.
+	const allVisual = useSelect( ( select ) => {
```

Comment:
One thing that we can consider for the performance of determining if `isEnabled` is maybe to 
* first check the `editorMode` and return early if it's not on `visual` and only if it is to check all the blocks. 
* when checking all blocks, return early when we first see a block mode that's not `visual`

so that we don't parse through all blocks and inner blocks every time.

But if we think that's a minor improvement and want to be pragmatic, I'm also ok with that 😅

Reply:
I like the first suggestion, I'll implement that 👍 

But `every` already returns early so the second point is not an issue 😄

Reply:
Resolved by putting the check in a single `useSelect` call, as also advised by https://developer.wordpress.org/news/2024/03/28/how-to-work-effectively-with-the-useselect-hook/#it-s-more-subtle-when-selecting-from-multiple-stores